crates/cairo-lint-core/src/lints/bool_assert_comparison.rs

```diff
@@ -0,0 +1,75 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::SyntaxNode;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use std::str::FromStr;
+
+pub const ASSERT_EQ_FALSE_MSG: &str = "assert_eq!(expr, false) can be replaced with assert!(!expr).";
+pub const ASSERT_EQ_TRUE_MSG: &str = "assert_eq!(expr, true) can be replaced with assert!(expr).";
+pub const ASSERT_NE_FALSE_MSG: &str = "assert_ne!(expr, false) can be replaced with assert!(expr).";
+pub const ASSERT_NE_TRUE_MSG: &str = "assert_ne!(expr, true) can be replaced with assert!(!expr).";
+
+fn evaluate_expression(expr: &str) -> Option<bool> {
+    bool::from_str(expr).ok()
+}
+
+pub fn check_assert(db: &dyn SyntaxGroup, node: SyntaxNode, diagnostics: &mut Vec<PluginDiagnostic>) {
+    let cloned_node = node.clone();

```

Comment:
do this `ExprInlineMacro::from_syntax_node(node)` so then you can use the provided methods for a macro and it'll avoid you doing manual stuff to extract the text